### ISSUE TITLE
CHALLENGE: Exclude Fields in Survey Being Submitted

### ISSUE THREAD

**Issue by phil-wilson** [tedrick/Survey123-Testing#9]
**Context:** In version 2.4 of Survey123 we've added new functionally to allow fields in the survey form to be excluded from being created/existing in the feature service when publishing from Connect. Additionally when collecting data in the field app the values entered in these fields will be excluded form being submitted to the feature service. These excluded fields will behave in the field app just like regular fields meaning they can contain calculations, relevant and constraints and be part of other calculations in other questions in the survey.

**Challenge:** Create and publish a survey in Connect with excluded fields using your own survey examples that include different question types and/or repeats and test against ArcGIS Online or Portal.

- Ensure the bind::esri:fieldType column exists in Survey sheet in XLS.
- Use the 'null' parameter in bind::esri:fieldType column against different field types and publish the survey.
- After publishing the survey check the field is not created in feature service in ArcGIS Online or Portal.
- Download the survey and submit data via the field app and ensure all values are displayed as expected and when sending that the values are updated and excluded as intended.
- Further test with existing feature services by adding new fields to survey with 'null' property that do not exist and test that the survey can be published and surveys sent from field app.
- Test constraints, relevant, calculations based on previous questions, and also calculations based on calculations in 'null' field types etc

**Note:** The following question types are supported with 'null' parameter:

- integer
- decimal
- text
- select_one [list_name]
- select_multiple [list_name]
- note
- date
- time
- dateTime
- calculate
- username
- email
- barcode
- start
- end

The following question types are **NOT** supported:

- select_one [list_name] or other
- select_multiple [list_name] or other
- geopoint
- image
- hidden

**Comment by tedrick** [comment]
Britney

**Comment by gibbonsbr** [comment]
I tested using a copy of an existing survey in Survey123 Connect version 2.4.32 (beta). I assigned a null field type to four fields that are used to hold intermediate calculations. A fifth column contains a final calculation that is recorded and this final calculation is based on the results of the intermediate calculations in the previous four fields. I experienced no problems publishing the survey, downloading the form, submitting a completed survey, or editing a completed survey.
